app/modules/intelligence/tools/tool_service.py

```
@@ -65,8 +66,18 @@ def _initialize_tools(self) -> Dict[str, Any]:
             "get_node_neighbours_from_node_id": GetNodeNeighboursFromNodeIdTool(
                 self.db
             ),
+            "ollama_tool": Ollama(
+                base_url=self._get_ollama_endpoint(),
+                model=self._get_ollama_model(),
+            ),
         }
 
+    def _get_ollama_endpoint(self) -> str:
+        return self.db.query(ConfigProvider).first().get_ollama_config()["endpoint"]
```

Ruff (0.8.2)

Undefined name `ConfigProvider`

(F821)